modules/300-prometheus/images/prometheus/sample_limit_annotation.patch

```diff
@@ -1,32 +1,40 @@
 diff --git a/discovery/kubernetes/pod.go b/discovery/kubernetes/pod.go
-index 2e55dce78..e84fa50b0 100644
+index 396720c22..faf28654a 100644
 --- a/discovery/kubernetes/pod.go
 +++ b/discovery/kubernetes/pod.go
-@@ -279,6 +279,7 @@ func (p *Pod) buildPod(pod *apiv1.Pod) *targetgroup.Group {
+@@ -299,6 +299,7 @@ func (p *Pod) buildPod(pod *apiv1.Pod) *targetgroup.Group {
+ 			// The user has to add a port manually.
+ 			tg.Targets = append(tg.Targets, model.LabelSet{
+ 				model.AddressLabel:     lv(pod.Status.PodIP),
++				"__sample_limit__":     lv(""),
+ 				podContainerNameLabel:  lv(c.Name),
+ 				podContainerIDLabel:    lv(cID),
+ 				podContainerImageLabel: lv(c.Image),
+@@ -313,6 +314,7 @@ func (p *Pod) buildPod(pod *apiv1.Pod) *targetgroup.Group {
 
  			tg.Targets = append(tg.Targets, model.LabelSet{
  				model.AddressLabel:            lv(addr),
 +				"__sample_limit__":            lv(""),
  				podContainerNameLabel:         lv(c.Name),
- 				podContainerPortNumberLabel:   lv(ports),
```

Comment:
Why have we deleted these labels?

Reply:
This is not deleted, the line is changed from 
```
podContainerPortNumberLabel: lv(ports),
```
to
```
podContainerIDLabel: lv(cID),
```
but not deleted.

See 
![image](https://user-images.githubusercontent.com/32434187/231681619-e15a9d18-d5be-4b68-94c0-c130ad1321d2.png)

The + in the picture is connected to the patch, but the minus is connected to github diff.